control-plane-operator/hostedclusterconfigoperator/controllers/resources/resources.go

```diff
@@ -2299,6 +2299,12 @@ func (r *reconciler) reconcileImageContentPolicyType(ctx context.Context, hcp *h
 		return fmt.Errorf("failed to reconcile image digest mirror set: %w", err)
 	}
 
+	if (idms.Spec.ImageDigestMirrors != nil && len(idms.Spec.ImageDigestMirrors) <= 0) || idms.Spec.ImageDigestMirrors == nil {
```

Comment:
nit: we don't need idms.Spec.ImageDigestMirrors != nil check given idms.Spec.ImageDigestMirrors == nil is in the other side of the OR.

Let's keep it for consistency since this is a backport though.

Reply:
Makes sense, I will change it in one of the refactor tasks